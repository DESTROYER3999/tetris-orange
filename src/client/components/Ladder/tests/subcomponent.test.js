import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import LadderSub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const props = {
  highscores: [
    { score: 10500, playerName: "Player" },
    { score: 10500, playerName: "Player" },
    { score: 10500, playerName: "Player" },
    { score: 10500, playerName: "Player" },
    { score: 10500, playerName: "Player" }
  ]
};

describe("<LoginSub />", () => {
  test("LadderSub rendering", () => {
    const wrapper = render(<LadderSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
