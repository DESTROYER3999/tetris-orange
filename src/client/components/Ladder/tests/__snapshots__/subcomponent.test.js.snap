// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> LadderSub rendering 1`] = `
<div
  class="sc-bdVaJa cSBjYp"
  id="ladder"
>
  <h2>
    High Scores
  </h2>
  <div
    class="sc-bwzfXH ikfEIr"
    id="highscore-list"
  >
    <div
      class="sc-htpNat hukHjT"
      id="list-item"
    >
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          1.
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          10500
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          Player
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cyWvph"
      id="list-item"
    >
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          2.
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          10500
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          Player
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat hukHjT"
      id="list-item"
    >
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          3.
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          10500
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          Player
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cyWvph"
      id="list-item"
    >
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          4.
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          10500
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          Player
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat hukHjT"
      id="list-item"
    >
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          5.
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          10500
        </div>
      </div>
      <div
        class="sc-bxivhb hpGinl"
        id="content"
      >
        <div>
          Player
        </div>
      </div>
    </div>
  </div>
</div>
`;
